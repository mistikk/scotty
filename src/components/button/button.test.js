import React from 'react';
import renderer from 'react-test-renderer';

import Button from './button.view';

describe('Button Component', () => {
  it('> render the component', () => {
    const component = renderer.create(<Button text="test" />);

    expect(component).toMatchSnapshot();
  });

  it('> button onClick function', () => {
    const mockCallback = jest.fn();
    const component = renderer.create(<Button text="test" onPress={mockCallback} />);

    component.root.findByType(Button).props.onPress();
    expect(mockCallback.mock.calls.length).toEqual(1);
  });
});
