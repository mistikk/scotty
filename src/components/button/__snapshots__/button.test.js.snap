// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component > render the component 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    test
  </Text>
</View>
`;
